lib/puppet/face/config.rb

```diff
@@ -106,6 +106,19 @@
     EOT
 
     when_invoked do |name, value, options|
+      unless name == 'environment' && (options[:section] == 'agent' || options[:section] == 'user')
```

Comment:
I would also allow `master`. We also want to point out why this is recommended: different modes use environment differently, the `puppet agent` doesn't require a local directory environment but most other operations do.

Reply:
Updated